clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -98,16 +113,71 @@ public void onFailure(RuntimeException e) {
         return fetchRequestMap.size();
     }
 
-    public void close(final Timer timer) {
-        if (!isClosed.compareAndSet(false, true)) {
-            log.info("Fetcher {} is already closed.", this);
-            return;
+    public Fetch<K, V> collectFetch() {
+        return fetchCollector.collectFetch(fetchBuffer);
+    }
+
+    public boolean hasCompletedFetches() {
```

Comment:
Looks to me this is only used for testing so maybe leave it package-private as it was before and add the visibleForTesting comment

Reply:
`Fetcher.hasCompletedFetches()` is is used from `KafkaConsumer.poll()`. The former is in the `o.a.k.clients.consumer.internals` package and the latter is in the `o.a.k.clients.consumer` package. Since they weren't the same package I assumed that the method needed to be `public`. Weird 🤔

Reply:
I removed the `public` method qualifier, as suggested. Thanks!